src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt

```diff
@@ -594,7 +597,8 @@ internal class RecommendationService(
     val legacyRecommendationOpen = recommendationEntity.size > 1
     val legacyStatusOpen = recommendationEntity.isNotEmpty() &&
       recommendationStatusRepository.findByRecommendationId(recommendationEntity[0].id)
-        .any { it.active && (it.name != "CLOSED" && it.name != "DELETED") }
+        .filter { it.active }
+        .none { it.name == "CLOSED" || it.name == "DELETED" }
```

Comment:
Are you sure this is doing the same thing as the .any?

Could be wrong here, but I think the any is doing:
return true if any of the items are active but not closed and not deleted

and the filter and none combination is doing:
return true **if there are no active items** or there are active items and none of them are closed or deleted

Reply:
see below :)